nested-spire/README.md

```diff
@@ -0,0 +1,226 @@
+       # Share the host pid namespace so this agent can attest the nested servers
+       pid: "host"
+       image: gcr.io/spiffe-io/spire-agent:0.10.1
+       depends_on: ["root-server"]
+       hostname: root-agent
+       volumes:
+         # Share root agent socket to be accessed by nestedA and nestedB servers
+         - ./sharedRootSocket:/opt/spire/sockets
+         - ./root/agent:/opt/spire/conf/agent
+         - /var/run/:/var/run/
+       command: ["-config", "/opt/spire/conf/agent/agent.conf"]
+```
+
+## Configuring NestedA SPIRE Deployment
+
+The same set of configurations are required for the `nestedB` SPIRE deployment but those changes are not described in the text to avoid needless repetition.
+
+SPIRE Agent and Server can be extended with a variety of [plugins](https://spiffe.io/spire/docs/extending/). The [UpstreamAuthority plugin](https://github.com/spiffe/spire/blob/master/doc/spire_server.md#built-in-plugins) type allows SPIRE Server to integrate with existing PKI systems. UpstreamAuthority plugins can sign certificates using CAs loaded from disk, third-party tools from AWS and Vault, and so on. Nested SPIRE deployments require the use of the [spire UpstreamAuthority plugin](https://github.com/spiffe/spire/blob/master/doc/plugin_server_upstreamauthority_spire.md) which uses an upstream SPIRE Server in the same trust domain to obtain intermediate signing certificates for SPIRE Server.
+
+The configuration file for the [nestedA-server](./nestedA/server/server.conf) includes the `spire` UpstreamAuthority plugin definition with the `root-server` as its upstream SPIRE Server.
+
+```console
+   UpstreamAuthority "spire" {
+ 	   plugin_data = {
+ 	       server_address      = "root-server"
+ 	       server_port         = 8081
+ 	       workload_api_socket = "/opt/spire/sockets/workload_api.sock"
+ 	   }
+    }
+```
+
+The Docker Compose definition for the `nestedA-server` service in the [docker-compose.yaml](docker-compose.yaml) file mounts the new local directory `sharedRootSocket` as a volume. Remember from the previous section that the `root-agent` socket is mounted on that directory. That way the `nestedA-server` can access the `root-agent` workload API and fetch its SVID.
+
+```console
+   nestedA-server:
+     # Share the host pid namespace so this server can be attested by the root agent
+     pid: "host"
+     image: gcr.io/spiffe-io/spire-server:0.10.1
+     hostname: nestedA-server
+     labels:
+       # label to attest nestedA-server against root-agent
+       - org.example.name=nestedA
+     volumes:
+       # Add root agent socket  
+       - ./shared/rootSocket:/opt/spire/sockets
+       - ./nestedA/server:/opt/spire/conf/server
+     command: ["-config", "/opt/spire/conf/server/server.conf"]
+```
+
+## Create Downstream Registration Entry
+
+The `nestedA-server` must be registered on the `root-server` to get SVIDs. We achieve this by creating a registration entry in the root SPIRE Server for the `nestedA-server`.
```

Comment:
why it is created? can you explain that this done in order to provide an identity to `nestedA-server` that will be used by plugin in order to mint svids?

Reply:
and that `-spiffeID "spiffe://example.org/nestedA"` is the server's id